src/amber/router/cookies/store.cr

```diff
@@ -0,0 +1,128 @@
+require "./*"
+require "http"
+
+module Amber::Router::Cookies
+  class Store
+    include Enumerable(String)
+
+    getter cookies
+    getter secret : String
+    property host : String?
+    property secure : Bool = false
+
+    def initialize(@host = nil, @secret = Random::Secure.urlsafe_base64(32), @secure = false)
+      @cookies = {} of String => String
+      @set_cookies = {} of String => HTTP::Cookie
+      @delete_cookies = {} of String => HTTP::Cookie
+    end
+
+    def self.from_headers(headers)
```

Comment:
This seems to be a constructor but instead it returns hash of `HTTP::Cookie`s.

Reply:
To be honest, I still find a lot of this code really confusing. Why is there a base Store class, but then we have other actual implementations of stores? I'd love to address further improvement here in other PR where I suspect we'd have some larger breaking changes.

Reply:
Totally agree, that piece of the code screams for refactoring - in another PR that is.